### PR TITLE
Adding "status", "commit", and "create-arrayset" to CLI

### LINKED ISSUE #132: [FEATURE REQUEST] More options through command line interface
**Is your feature request related to a problem? Please describe.**
We need more options in our CLI for making few workflows seamless. Two examples I could think off
1. create dataset: We have repo init and data import through CLI but without the option to create a dataset through CLI, the user will always have to go through python APIs
2. commit: Now, if users import data using `hangar import`, they will need to spin up a python program and use the python APIs to commit the staged data

### PR DESCRIPTION
**Pull request by rlizzo**
#### _Why is this change required? What problem does it solve?:_

Adding the ability to perform `status`, `commit`, and `create arrayset` operations from the command line interface. 

#### _If it fixes an open issue, please link to the issue here:_
 
- closes #132

#### _Describe your changes in detail:_

WIP, will update if necessary

## Types of changes
What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Documentation update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Is this PR ready for review, or a work in progress?
- [x] Ready for review
- [ ] Work in progress

## How Has This Been Tested?
Put an `x` in the boxes that apply:
- [x] Current tests cover modifications made
- [x] New tests have been added to the test suite
- [ ] Modifications were made to existing tests to support these changes
- [ ] Tests may be needed, but they are not included when the PR was proposed
- [ ] I don't know. Help!

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTING](../CONTRIBUTING.rst)** document.
- [x] I have signed (or will sign when prompted) the tensorwork CLA.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/hangar/cli/cli.py [hunk 1]
```diff
@@ -52,6 +53,158 @@ def init(repo: Repository, name, email, overwrite):
...
+            pass
+
+
+@arrayset.command(name='remove')
+@click.argument('name', nargs=1, type=click.STRING, required=True)
+@pass_repo
+def remove_arrayset(repo: Repository, name):
+    """Delete the arrayset NAME (and all samples) from staging area.
+
+    The arrayset will be created in the staging area / branch last used by a
```
Comment (hhsecond) [inline src/hangar/cli/cli.py]: created -> removed

File: src/hangar/cli/cli.py [hunk 2]
```diff
@@ -52,6 +53,158 @@ def init(repo: Repository, name, email, overwrite):
...
+@main.group(no_args_is_help=True, add_help_option=True)
+@click.pass_context
+def arrayset(ctx):  # pragma: no cover
+    """Operations for working with arraysets in the writer checkout.
+    """
+    pass
+
+
+@arrayset.command(name='create')
+@click.option('--variable-shape', '-vs', 'variable_', is_flag=True, default=False,
```
[the remark this answers is not in the document]
Comment (rlizzo) [reply src/hangar/cli/cli.py]: good point. any suggestions?

[the remark this answers is not in the document]
Comment (hhsecond) [reply src/hangar/cli/cli.py]: `-v` is not been used in our CLI yet, so that's there. Or `--vs` perhaps?

[the remark this answers is not in the document]
Comment (rlizzo) [reply src/hangar/cli/cli.py]: `-v` it is.

[the remark this answers is not in the document]
Comment (elistevens) [reply src/hangar/cli/cli.py]: Shouldn't `-v` be verbose?

[the remark this answers is not in the document]
Comment (rlizzo) [reply src/hangar/cli/cli.py]: yup. I just killed the short version... It's really not necessary for a flag option. typing `--variable-shape` is doable.

**Pull request merged**